app/controllers/spree/admin/product_imports_controller.rb

```diff
@@ -1,20 +1,31 @@
 class Spree::Admin::ProductImportsController < Spree::Admin::BaseController
+  before_action :initialize_resource, only: [:new]
+  before_action :load_resource, only: [:create]
 
-  def index
-    @product_import = Spree::ProductImport.new
+  def new
   end
 
   def create
-    @product_import = Spree::ProductImport.new(product_import_params.merge(user_id: spree_current_user.id))
     if @product_import.save
-      redirect_to admin_product_imports_url, notice: "Import process started successfully"
+      redirect_to new_admin_product_import_path, notice: Spree.t(:success, scope: [:admin, :product_import])
     else
-      render :index
+      flash.now[:error] = Spree.t(:error, scope: [:admin, :product_import])
+      render :new
     end
   end
 
   private
     def product_import_params
-      params.require(:product_import).permit(:products_csv)
+       product_import_params = params.require(:product_import).permit(:products_csv)
+       product_import_params.merge(user_id: spree_current_user.id)
     end
+
+    def initialize_resource
```

Comment:
@AnuragBhardwaj1 - No need to create two different methods, we can directly use them in new and create actions.